File: Week 6/cash.py
from cs50 import get_float


def main():
    number = 0
    while (number <= 0):
        number = get_float("Enter your cash: ")
    print(cash(number))


def cash(number):
    coins = 0
    while number >= 1:
        number -= 1
        coins += 4
    while number >= 0.25:
        number -= 0.25
        number = round(number, 2)
        coins += 1
    while number >= 0.1:
        number -= 0.1
        number = round(number, 2)
        coins += 1
    while number >= .05:
        number -= 0.05
        number = round(number, 2)
        coins += 1
    while number >= 0.01:
        number -= 0.01
        number = round(number, 2)
        coins += 1
    return coins


main()

File: Week 1/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int get_change(int cash);

int main(void)
{
    float cash;
    int counter = 0;
    do
    {
        cash = get_float("Change owed: ");
    }
    while (cash <= 0);
    int cents = round(cash * 100);
    counter = get_change(cents);
    printf("%i\n", counter);
}

int get_change(int cash)
{
    int counter = 0;
    while (cash > 0)
    {
        if (cash >= 25)
        {
            cash -= 25;
            counter++;
        }
        else if (cash < 25 && cash >= 10)
        {
            cash -= 10;
            counter++;
        }
        else if (cash < 10 && cash >= 5)
        {
            cash -= 5;
            counter++;
        }
        else if (cash < 5)
        {
            cash--;
            counter++;
        }
    }
    return counter;
}
